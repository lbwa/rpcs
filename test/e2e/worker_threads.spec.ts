import path from 'path'
import { Worker } from 'worker_threads'

describe('worker_threads', () => {
  let handler: any
  let worker: Worker
  beforeAll(() => {
    worker = new Worker(
      path.resolve(__dirname, './fixtures/worker_threads/worker.js')
    )
    handler = require('../../dist/index.cjs').wrapRpc(worker)
  })

  afterAll(async () => {
    await worker?.terminate()
    handler = null
  })

  it('should get state calling', async () => {
    expect(await handler.primitive).toEqual(1)
    expect(await handler.object).toEqual({ name: 'worker name' })
    expect(await handler.promise).toBeUndefined()
  })

  it('should get method calling', async () => {
    expect(await handler.getNumber(1)).toEqual(1)
    expect(await handler.getNumber()).toEqual(12)
    expect(await handler.returnPromise(2)).toEqual(2)
    expect(await handler.returnPromise()).toEqual(12)
    await expect(async () => {
      await handler.returnReject()
    }).rejects.toMatchSnapshot('catch reject promise')
  })

  it('should catch unavailable calling', async () => {
    await expect(async () => {
      await handler.randomCall()
    }).rejects.toMatchSnapshot('catch unavailable calling')
    await worker.terminate()
  })
})
