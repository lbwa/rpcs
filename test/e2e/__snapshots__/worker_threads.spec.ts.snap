// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`worker_threads should catch unavailable calling: catch unavailable calling 1`] = `[Error: <REMOTE>.randomCall isn't a callable function.]`;

exports[`worker_threads should get method calling: catch reject promise 1`] = `"rejected promises from worker thread"`;
